components/import/basic.tsx

```diff
@@ -69,32 +74,44 @@ function Basic() {
       }}
       onSubmit={async values => {
         setIsLoading(true);
-        const contract = await state.connection.contract.at(
-          values.walletAddress,
-          tzip16
-        );
-        const storage: any = await contract.storage();
-        let version = await fetchVersion(contract!);
+        setError(undefined);
 
-        let balance = await state?.connection.tz.getBalance(
-          values.walletAddress
-        );
-        let v = toStorage(version, storage, balance);
-        const validators = signers(v).map((x: string) => ({
-          address: x,
-          name: state.aliases[x] || "",
-        }));
+        try {
+          const contract = await state.connection.contract.at(
+            values.walletAddress,
+            tzip16
+          );
+          const storage: any = await contract.storage();
+          let version = await fetchVersion(contract!);
 
-        const data = {
-          ...formState,
-          ...values,
-          validators,
-          requiredSignatures: storage.threshold.toNumber(),
-          effectivePeriod: storage.effective_period.toNumber(),
-        };
-        setFormState(data as any);
-        setActiveStepIndex(activeStepIndex + 1);
-        setIsLoading(false);
+          if (version === "unknown version") {
+            throw new Error("The contract is not a TzSafe contract");
+          }
+
+          let balance = await state?.connection.tz.getBalance(
+            values.walletAddress
+          );
+          let v = toStorage(version, storage, balance);
```

Comment:
is it possible to have errors here? if yes, we should handle it.

Reply:
Everything is in a try/catch so it's already handled